rope/contrib/generate.py

```diff
@@ -42,15 +42,16 @@ def create_package(project, name, sourcefolder=None):
 
 class _Generate(object):
 
-    def __init__(self, project, resource, offset):
+    def __init__(self, project, resource, offset, goal_resource=None):
```

Comment:
It should be invalid for `GenerateModule` or `GeneratePackage`, there's nothing additional to do for these. However, being as they override `get_changes`, maybe this doesn't matter since it would just be ignored.

Reply:
I admit that it is poor style to declare fields in parent class that are not used in child class. I have chosen it because it needs the smallest changes.

As an alternative I could change hierarchy so that it would look something like:
- _ModuleElementGenerate(_Generate)
- GenerateClass(_ModuleElementGenerate)
- GenerateFunction(_ModuleElementGenerate)
- GenerateVariable(_ModuleElementGenerate)
- GenerateModule(_Generate)
- GeneratePackage(_Generate)

Should I change it?

Reply:
No, I agree it's better as-is, and won't cause any harm. Thanks!